src/Illuminate/Validation/Validator.php

```diff
@@ -508,7 +515,13 @@ public function validated()
 
         $missingValue = new stdClass;
 
-        foreach (array_keys($this->getRules()) as $key) {
+        foreach ($this->getRules() as $key => $rules) {
+            if ($this->excludeArrays &&
+                in_array('array', $rules) &&
+                ! empty(preg_grep('/^'.$key.'\.*/', array_keys($this->implicitAttributes)))) {
```

Comment:
Escaping the regex pattern with `preg_quote($key, '/')` may be required to cover every character that `<input name="">` allows, not just restricted to alphanumeric/underscore.

e.g., validating a parsed XML document (external spec) can lead to some weird field name edge cases with hyphens, slashes, etc.